src/viewtransaction/viewTransaction.es6

```diff
@@ -350,7 +350,7 @@ function drawXLines(contract, state, chart) {
     }
 
     if (is_sold_before_expiry) drawXLine({ line_time: sell_time, label: 'end_time', dashStyle: 'Dash' });
-    if (!is_path_dependent) drawXLine({ line_time: date_expiry, label: 'end_time', dashStyle: 'Dash' });
+    if (!is_path_dependent) drawXLine({ line_time: date_expiry && exit_tick_time, label: 'end_time', dashStyle: 'Dash' });
```

Comment:
shouldn't it be `if (!is_path_dependent && date_expiry)`?

Reply:
indeed. when i debug it the problem was, every tick should have date_expiry carry along. but then some ticks are not consistent when it stops and date expiry is not exist. thats when the problem happens. it depends on internet connection, sometime i got all date expiry carry along, sometimes some tick dont have date expiry. but by default, they should have date expiry. so i made a fallback if date expiry is not exist then use exit tick time instead. what do you think